storybook/qmlTests/TestComponents/tst_StatusChatInput.qml

```diff
@@ -1,21 +1,571 @@
+                verify(controlUnderTest.textInput.text === controlUnderTest.getTextWithPublicKeys(),
+                                            `Expected text: ${controlUnderTest.textInput.text}, received: ${controlUnderTest.getTextWithPublicKeys()}`)
+            }
+
+            testHelper.when_text_is_typed(statusChatInputKeyboardInputExpectedAsText,
+                                          testHelper.get_all_ascii_characters(),
+                                          expectations_after_typed_text);
+        }
+
+//        Scenario outline: User can navigate in chat input using keyboard
+//        Given the user has typed text on multiple lines in StatusChatInput
+//        """
+//        Typing on first row
+//        Typing on the second row
+//        Typing on the third row
+//
+//        """
+//        When the user hits <direction>
+//        The cursor will move inside the text area to the <direction>
+//
+//        Examples:
+//        |direction|
+//        |left|
+//        |right|
+//        |up|
+//        |down|
+
+//        When the user hits "left"
+//        The cursor will move inside the text area to the "left"
+//        When the user hits "right"
+//        The cursor will move inside the text area to the "right"
+//        When the user hits "up"
+//        The cursor will move "up" inside the text area, on the previous line if any
+//        When the user hits "down"
+//        The cursor will move "down" inside the text area, on the next line if any
+        function test_user_can_navigate_with_keys() {
+
+            const firstRowText = "Typing on first row"
+            testHelper.when_text_is_typed(statusChatInputKeyboardInputExpectedAsText,
+                                          firstRowText, (typedText) => { })
+            keyClick(Qt.Key_Enter, Qt.ShiftModifier)
+            compare(controlUnderTest.getPlainText(),
+                    firstRowText + "\n", controlUnderTest.getPlainText())
+
+            const secondRowText = "Typing on the second row"
+            testHelper.when_text_is_typed(statusChatInputKeyboardInputExpectedAsText,
+                                          secondRowText, (typedText) => { })
+            keyClick(Qt.Key_Enter, Qt.ShiftModifier)
+            compare(controlUnderTest.getPlainText(),
+                    firstRowText + "\n" + secondRowText + "\n", controlUnderTest.getPlainText())
+
+            const thirdRowText = "Typing on the third row"
+            testHelper.when_text_is_typed(statusChatInputKeyboardInputExpectedAsText,
+                                          thirdRowText, (typedText) => { })
+            keyClick(Qt.Key_Enter, Qt.ShiftModifier) //Add an empty 4th line
+            compare(controlUnderTest.getPlainText(),
+                    firstRowText + "\n" + secondRowText + "\n" + thirdRowText + "\n", controlUnderTest.getPlainText())
+
+            compare(controlUnderTest.textInput.lineCount, 4,
```

Comment:
If you use the multiline method, maybe you can parametrize the `4` number by reading the array length!

Reply:
Ohh, yes. Looks like I forgot to use the multiline function everywhere. Will update